Use default editor to write comments and descriptions if specified

kimpers opened the pull request:
Descriptions and comments can be quite long and cumbersome to input in the default input prompt so therefore it's better to use the system default $EDITOR for composing the content.

This feature was requested in #40.
Feel free to comment on code style, it has been a while since I developed with Ruby

keepcosmos commented:
Thanks for PR! First of all, Opening editor(it's what we need) is good for writing and editing long text. 

But I think, the default should be inline or prompt. Opening editor should be optional (pass option arg or global configuration).

kimpers commented:
That is a good idea. I have updated the pr to accept `-e` or `--editor` to open comment or description in the default editor

keepcosmos commented:
Thanks :) ,

There are 2 problem I found.
First, option `-e` is used for add Epic link. And Add to `jira_options`. That is above method definition. 
Second, It needs fallback handling if $EDITOR env is not set.

Please fix 2 issues before I merge your nice code. :)

kimpers commented:
Thank you for the feedback. I didn't understand what `jira_options` did initially.

I have cleaned up the code a bit, fixed so that it raises an error if you try to edit with an editor without having $EDITOR set to something or if the editor returns a non-zero status code. 

I also changed the alternative argument for using an editor to `-E`

keepcosmos commented:
Good, but you should fix one more thing. `editor` variable is not a resource of jira. I mean that `editor` boolean variable makes error when parsing jira resource.  

So, you should handle `editor` option another way. 

For example,

```
suggest_options(required: [:project, :summary, :issuetype],
                             editor: options.editor)
```

And please check your code works :)

kimpers commented:
I apologize for breaking the code. I was manually testing the `comment` command and forgot to check `new` & `edit`. I will fix this as soon as possible

Regarding not passing it in resources. The reason for passing it as a resource was because it was a nice way to have it accessable in `option_selector.rb`.

How would you propose passing the value forward through `option_supportable` (https://github.com/kimpers/terjira/blob/08bccbb2f3c527ca6bb00ecb1743e3e0a7f7f6d2/lib/terjira/option_supportable.rb#L66) without having to modify the `selector_method` to take arguments?

I am apologize if this a stupid question, it's been a very long time since I did Ruby development 🙃

kimpers commented on lib/terjira/option_supportable.rb:
```diff
@@ -30,8 +30,9 @@ def self.included(klass)
+      self.with_editor = options[:editor]
       origin = options.dup
-      self.with_editor = opts[:editor]
+      origin.delete(:editor)
```
This was causing the issues earlier. Other logic that parses the options in the client couldn't handle the that the `editor` option was in the hash.

kimpers commented on lib/terjira/option_supportable.rb:
```diff
@@ -30,7 +30,9 @@ def self.included(klass)
     def suggest_options(opts = {})
+      self.with_editor = options[:editor]
       origin = options.dup
+      origin.delete(:editor)
```
This was causing the issues earlier. Other logic that parses the options in the client couldn't handle the that the editor option was in the hash.

kimpers commented:
I have addressed your comments and the issues should be fixed now.

I think travis are having issues right now which is why one of the jobs is failing 😞

kimpers commented:
Retriggering travis does not seem to help

keepcosmos commented:
I just merge your code and release 0.3.7. Thanks! ❤️ 
Please download the version and check! 👍
